core/io/compression.cpp

```diff
@@ -57,6 +61,22 @@ int Compression::compress(uint8_t *p_dst, const uint8_t *p_src, int p_src_size,
 				return fastlz_compress(p_src, p_src_size, p_dst);
 			}
 
+		} break;
+		case MODE_LZ4: {
+#ifdef LZ4_ENABLED
+			if (p_src_size < 16) {
+				uint8_t src[16];
+				memset(&src[p_src_size], 0, 16 - p_src_size);
+				memcpy(src, p_src, p_src_size);
+				return LZ4_compress_default(reinterpret_cast<const char *>(src), reinterpret_cast<char *>(p_dst),
+						p_src_size, LZ4_compressBound(p_src_size));
+			}
```

Comment:
Is this part necessary? `fastlz` have it since it specify minimum input size, but I do not see anything in the LZ4 header.

Reply:
So that's the reason thanks! I had it there because FastLZ had it, I `git blame`d trying to get the why but couldn't find any info so I supposed it was some optimization trick. As FastLZ is similar to LZ4, I added it there too.
Anyway, you're right here, just removed it.